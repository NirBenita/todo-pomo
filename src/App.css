.App {
  background-image: linear-gradient(-137deg, #223855 0%, #202652 100%);
  border-radius: 8px;
  width: 276px;
  overflow: hidden;
  color: white;
  font-family: HelveticaNeue-Light;
  box-shadow: 0 2px 12px 0 rgba(0, 0, 0, 0.36);
  margin: 24px auto;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

header {
  display: flex;
  text-align: center;
  justify-content: center;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.3);
  padding: 0px 24px;
}

header > div {
  padding: 12px 24px;
}

.content {
  padding: 0 24px 18px 24px;
}

.shiny {
  background-image: linear-gradient(-90deg, #8746ff 1%, #00cbff 100%);
  height: 2px;
  width: 100%;
}

h4 {
  font-size: 14px;
  line-height: 48px;
}

.pill {
  background: #ffffff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.48);
  border-radius: 6px;
  overflow: hidden;
  color: #4a4a4a;
  line-height: 36px;
  outline: none;
  border: none;
  /* padding: 0 12px; */
  width: 100%;
  height: 36px;
  transition: .2s background-color ease-in;
  transition: .02s flex-basis ease-in;
  /* border-bottom: 2px solid #f2f2f2; */
  display: flex;
  flex-direction: column;
}

.pill:hover {
  background-color: #e8f1fd;
}

.todo {
  cursor: pointer;
  display: flex;
  align-items: center;
  margin: 0 12px;
}

.todo-toggle {
  margin-right: 6px;
  border: 1px solid red;
}

.todo-pom-counter {
  display: flex;
}
.todo-pom-indicator {
  height: 4px;
  width: 4px;
  border-radius: 99px;
  background-color: #c9d5f5;
  margin: 0 2px;
}
.todo-pom-indicator.done {
  background-image: linear-gradient(-145deg, #8746ff 0%, #00cbff 100%);
}
.todo-progress {
  display: flex;
  height: 2px;
  background-color: gray;
}
.todo-progress-done {}

.todo-label {
  flex: 1;
}

label {
  opacity: 0.6;
  font-family: HelveticaNeue;
  font-size: 12px;
  color: #8eabf5;
  line-height: 24px;
}

a {
  font-size: 14px;
  color: #c9d5f5;
  line-height: 36px;
  text-decoration: none;
}

a:hover {
  text-decoration: underline;
}

.action {
  color: #207cf0;
  font-weight: bold;
}
