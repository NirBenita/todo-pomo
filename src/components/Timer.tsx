import * as React from 'react'

interface TimerState {
  startTime: number
  endTime: number
  timeLeft: number
  timeInterval: NodeJS.Timer
}
interface TimerProps {
  time: number
  onComplete?: () => {}
}

export class Timer extends React.Component<TimerProps, TimerState> {
  constructor(props: TimerProps) {
    super(props)
  }

  updateTimeLeft() {
    const now: number = +new Date()
    const timeLeft = this.state.endTime - now

    if (timeLeft < 1) {
      clearInterval(this.state.timeInterval)
      this.props.onComplete && this.props.onComplete()
    }

    this.setState({ timeLeft })
  }

  intializeTimer(minutes: number) {
    const startTime: number = new Date().getTime()
    const endTime: number = new Date(startTime + minutes * 60 * 1000).getTime()

    this.setState({
      startTime,
      endTime
    })

    const timeInterval: NodeJS.Timer = global.setInterval(
      () => this.updateTimeLeft(),
      1000
    )
    this.setState({ timeInterval })
  }

  render() {
    return (
      <div>
        <button
          className="start"
          onClick={() => this.intializeTimer(this.props.time)}
        >
          Start Timer
        </button>
      </div>
    )
  }
}
