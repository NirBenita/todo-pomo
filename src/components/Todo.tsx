import * as React from 'react';

interface ITodo {
  title: string
}

interface TodosProps {
  todos?: Array<ITodo>
}

export const TodoList: React.SFC<TodosProps> = ({todos}) => 
  <div>
    <ul>
      {todos && todos.map((todo, index)=>
        <li key={index}>{todo.title}</li>
      )}
    </ul>
  </div>