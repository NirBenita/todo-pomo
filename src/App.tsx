import * as React from 'react'
import './App.css'
import Pill from './Pill'

/* TODO: 
  [ ] Timer progress animation
        [x] How to set flex basis at runtime?
  [ ] Add Todo
  [ ] Complete Todo
  [ ] Remove Todo
  - Pomadoro sessions indicator
    [x] skin
    [ ] logic

*/

export interface TimeType {
  total: string
}

interface AppState {
  timeLeft?: string
  completed: number
  intervalId?: NodeJS.Timer
}
class App extends React.Component<{}, AppState> {
  constructor(props: {}) {
    super(props)
    this.state = {
      completed: 0
    }
  }
  onClick() {
    this.initializeTimer(25)
  }
  getTimeRemaining(startTime: number, endTime: number) {
    // const t = Math.round(100-((endTime.getTime()-startTime.getTime()) * 100) / now ) +'%'
    var now: number = +new Date(),
      start: number = startTime * 1000,
      end: number = endTime * 1000

    const t = Math.round((now - start) / (end - start) * 100)
    console.log(Math.round((now - startTime) / (endTime - start) * 100))
    return t
  }

  updateTimer(startTime: number, endTime: number) {
    const timeRemaining = this.getTimeRemaining(startTime, endTime)

    this.setState({ completed: timeRemaining })
    // if (parseInt(timeRemaining.total) <= 0) {
    //   Stop
    // }
  }

  initializeTimer(timeInMinutes: number) {
    var startTime: number = new Date().getTime()
    var endTime: number = new Date(
      startTime + timeInMinutes * 60 * 1000
    ).getTime()

    if (this.state.intervalId) {
      clearInterval(this.state.intervalId)
    }
    this.updateTimer(startTime, endTime)

    let timeinterval = global.setInterval(
      () => this.updateTimer(startTime, endTime),
      100
    )
    this.setState({ intervalId: timeinterval })
  }

  render() {
    return (
      <div className="App">
        <header>
          <div>
            <h3>4</h3>
            <label>Weekly</label>
          </div>
          <div>
            <h3>0</h3>
            <label>Daily</label>
          </div>
        </header>
        <div className="shiny" />
        <section className="content">
          <h4>Main goal for today</h4>
          <Pill
            timer={true}
            onClick={() => this.onClick()}
            completion={50}
            intervalId={this.state.intervalId}
          />
          <span className="action">+ </span>
          <a href="#">Add a secondary task</a>
          <div />
        </section>
      </div>
    )
  }
}
export default App
