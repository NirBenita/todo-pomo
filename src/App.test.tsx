import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import App from './App'
import { TodoList } from './components/Todo'
// import mockTodoList from './lib/mock-data'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
})

it('should render todo list', () => {
  const wrapper = shallow(<TodoList />)
  
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('should display todos passed to it', () => {
  const mockTodos = [{ title: 'zagzag' }]

  const wrapper = mount(<TodoList todos={mockTodos} />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('should add a new todo', () => {
  const wrapper = mount(<TodoList />)

  expect(wrapper.find('.todos')).toHaveLength(0);

})