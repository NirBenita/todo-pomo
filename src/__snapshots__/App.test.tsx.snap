// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display todos passed to it 1`] = `
<Component
  todos={
    Array [
      Object {
        "title": "zagzag",
      },
    ]
  }
>
  <div>
    <ul>
      <li
        key="0"
      >
        zagzag
      </li>
    </ul>
  </div>
</Component>
`;

exports[`should render todo list 1`] = `
<div>
  <ul />
</div>
`;
